### PR TITLE
19-feature-enhancement-externalize-builder-function

### PR DESCRIPTION
**Pull request by uedar**
- Moved atomic structure creation part of build module that is common in STATE and QE to an external file 
- Wrote a simple pytest code on the changes I made.
- After merged, I will work on https://github.com/hhlim12/STATE-BATCH/issues/18

### CODE REVIEW

File: statebatch/statebatch/STATEbatch.py [hunk 1]
```diff
@@ -57,43 +53,10 @@ def link():
...
-                elif (crystalstructure+facet == 'bcc111'):
-                    atoms_obj = bcc111(_atoms, size=size, vacuum=vacuum)
-
-                if (self.system_spec.get('type') == 'Adsorption'):
-                    _adsorbate = atom_to_run['Adsorbate']
-                    adsorbate_obj = molecule(_adsorbate)
-                    add_adsorbate(atoms_obj, adsorbate_obj, height = atom_to_run['Height'], position = atom_to_run['Site'])
 
+        
+        def build(atom_to_run):
```
Comment (kimrojas) [inline statebatch/statebatch/STATEbatch.py:58]: @uedar There are no lines in `build` that will change any of the self attributes. This means that `build` can be fully treated as a external function with returns the tuple (atoms_obj, input_file, output_file). 
I think it is possible to further externalize the lines here to the `build_atoms_structure`.

Is it possible to also bring out the `build` module completely ?

File: tests/test_build.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+from statebatch.build import build_atoms_structure
```
Comment (kimrojas) [inline tests/test_build.py:1]: Can you please document how to run this testing codes ?

File: statebatch/statebatch/STATEbatch.py (hunk 1 above)
Comment (uedar) [reply statebatch/statebatch/STATEbatch.py]: @kimrojas Thanks for your help!

File: tests/test_build.py (hunk 2 above)
Comment (uedar) [reply tests/test_build.py:1]: I added description to readme.

**Review by kimrojas — COMMENTED** [review]
I corrected a syntax error bug and merged the patch. 
Its working now. 
This merge looks good.

**Review by kimrojas — APPROVED** [review]
OK for merge

**Pull request merged**